### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by levydsa**
Depends on #11

### CODE REVIEW

File: libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                )
+                .build()
+                .toByteArray()
+
+        return Rows(nativeQuery(this.inner, sql, buf))
+    }
+
+    override fun transaction(): Transaction {
+        require(this.inner != 0L) { "Transaction already closed" }
+        return Transaction(nativeTransaction(this.inner))
```
Comment (haaawk) [inline libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt:88]: Shouldn't this throw? I don't think nested transactions are supported.

**Review by haaawk — COMMENTED** [review]
LGTM modulo the ability to create nested transaction

File: libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt (hunk 1 above)
Comment (levydsa) [reply libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt:88]: If it is not supported, the error will already be thrown Rust

Comment (levydsa) [reply libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt:88]: I'll add a test case for this.

Comment (haaawk) [reply libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt:88]: Ideally we Transaction would not have `transaction()` method at all

**Comment by haaawk** [comment]
Why is Transaction implementing Connection? It seems somehow artificial.

**Comment by levydsa** [comment]
This is derived from Rust's behaviour, Transaction implements Deref<Target = Connection>, this is conceptualy making Transaction a subtype of Connection. It would be even more correct to make Transaction inherit Connection, in this sense.

**Comment by haaawk** [comment]
> This is derived from Rust's behaviour, Transaction implements Deref<Target = Connection>, this is conceptualy making Transaction a subtype of Connection. It would be even more correct to make Transaction inherit Connection, in this sense.

It feels like a mistake in the Rust SDK but ok let's keep it like this for now.

**Pull request merged**